tutorial-review.md

```diff
@@ -0,0 +1,1186 @@
+    "@testing-library/user-event": "^13.5.0",
+    "axios": "^0.26.0",
+    "bootstrap": "5.1.3",
+    "react": "^17.0.2",
+    "react-bootstrap": "^2.1.2",
+    "react-dom": "^17.0.2",
+    "react-hook-form": "^7.27.1",
+    "react-router-dom": "^6.2.1",
+    "react-scripts": "5.0.0",
+    "web-vitals": "^2.1.4"
+  },
+  "scripts": {
+    "start": "react-scripts start",
++   "build": "BUILD_PATH='../public' react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
+  "eslintConfig": {
+    "extends": [
+      "react-app",
+      "react-app/jest"
+    ]
+  },
+  "browserslist": {
+    "production": [
+      ">0.2%",
+      "not dead",
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  }
+}
+```
+
+If you want to verify that you setup everything correctly, take a look at project directory structure:
+
+```
+├── app.js
+├── bin
+│   └── www
+├── package.json
+├── public
+│   ├── images
+│   ├── javascripts
+│   └── stylesheets
+│       └── style.css
+├── routes
+│   ├── index.js
+│   └── users.js
+└── views
+    ├── error.pug
+    ├── index.pug
+    └── layout.pug
+└── client
+    ├── package.json
+    ├── yarn.lock
+    ├── public
+    └── src
+        ├── App.js
+        ├── App.css
+        ├── App.test.js
+        ├── index.js
+        ├── index.css
+        ├── logo.svg
+        ├── reportWebVitals.js
+        └── setupTests.js
+
+```
+
+### Create a database on Mongo Atlas
+
+The easiest way to setup our MongoDB database is to rely on a cloud service such as [Mongo Atlas](https://www.mongodb.com/cloud/atlas/lp/try2). They host databases on AWS, Google Cloud, Azure and provide several options to operate and scale MongoDB painlessly.
+
+
+In the plan page, choose the "shared" plan which starts for free:
+[mongo-plan]
+
+Next, choose the cloud provider and region, leave the default values for "cluster tier", "additional settings" and choose a cluster name, "mongo-blog-db".
+
+
+
+
+In the dashboard "quickstart" go ahead and create a username and password as we will need that to connect the database later on. Plus, we can restrict IP access to the database. This is very important measure when going to production with a real app but for the purpose of the tutorial we can simply add `0.0.0.0/0` to allow access from any IP.
+
+
+The database will successfully deploy in a few minutes, in the meantime we can click on the "connect" button, choose "connect your application" and copy the connection string.
+
+[mongo-connection]
+A typical connection string should look  like this:
+
+```
+mongodb+srv://samuele:<password>@mongo-client-db.r5bv5.mongodb.net/myFirstDatabase?retryWrites=true&w=majority
+```
+
+Move back the codebase, open `app.js` to require mongoose, connect it to the database by using the connection string and recover from potential errors:
+
+```javascript
```

Comment:
I am having an issue confirming my database becomes connected to the application and then when I try deploying on Koyeb I am receiving an authentication error. Did you have any issues with this when deploying to Koyeb?

Is there a way we could connect the database in the way it was done in this tutorial? https://www.koyeb.com/tutorials/deploy-a-graphql-api-with-mongodb-atlas-and-apollo-server-on-koyeb

Let me know what you think about this.

Reply:
hey @alisdairbr, I reviewed the tutorial and how I guide users to try the connection in local and then inject the connection string as env variable from Koyeb. You are right, I did overcomplicate things so I have just reworked it a little bit.

I do not add any nodemon related env variable but instead instruct the users to add the connection string inside `app.js`. Then, in the final part of the tutorial, when teaching readers to deploy on Koyeb, we are removing the hardcoded string and replace it with `process.env.CONNECTION_STRING` that we can inject. Please let me know if this fix your issue and it makes it clearer as well